e2e-pw/src/oss/poms/color-modal/index.ts

```diff
@@ -0,0 +1,127 @@
+import { Locator, Page, expect } from "src/oss/fixtures";
+
+export class ColorModalPom {
+  readonly page: Page;
+  readonly colorModal: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.colorModal = page.locator("#colorModal");
+  }
+
+  async closeColorModal() {
+    const closeButton = this.colorModal.getByTestId("close-color-modal");
+    return closeButton.click();
+  }
+
+  async selectActiveField(fieldName: string) {
+    const fieldSelector = this.page.getByTestId(
```

Comment:
can you introduce getter methods for all locators so we can reuse the locators across the page and abstract the locator logic into getters vs. actual testing logic

for example:
```
getFieldSelector(fieldName: string) {
  return this.page.getByTestId(`color-modal-list-item-${fieldName}`)
}
```

Now in the async calles, you can simply say

```
this.getFieldSelector().ACTION
```